File: crates/enforcer-memory/tests/fixtures/memory/lang_erlang/widget.erl
-module(widget).
-export([area/1, helper/1, draw/1]).
-import(lists, [sort/1]).

-type shape() :: {circle, float()} | {rectangle, float(), float()}.

helper(X) ->
    X + 1.

area({circle, R}) ->
    3.14 * R * R;
area({rectangle, W, H}) ->
    W * H.

draw(S) ->
    if
        area(S) > 0 -> io:format("visible~n");
        true -> io:format("hidden~n")
    end,
    helper(3).

File: crates/enforcer-memory/tests/fixtures/memory/lang_haskell/widget.hs
module Widget where

import Data.List (sort)
import qualified Data.Map as Map

data Shape = Circle Double | Rectangle Double Double

helper :: Int -> Int
helper x = x + 1

area :: Shape -> Double
area (Circle r) = pi * r * r
area (Rectangle w h) = w * h

draw :: Shape -> String
draw s =
  if area s > 0
    then "visible " ++ show (helper 3)
    else "hidden"
